Kernel_23/include/CGAL/Exact_predicates_inexact_constructions_kernel.h

```diff
@@ -19,28 +19,80 @@
 
 #include <CGAL/Simple_cartesian.h>
 #include <CGAL/Filtered_kernel.h>
+#include <CGAL/Converting_construction.h>
 #include <CGAL/Triangulation_structural_filtering_traits.h>
 
 namespace CGAL {
 
-// The following is equivalent to Filtered_kernel< Simple_cartesian<double> >,
-// but it's shorter in terms of template name length (for error messages, mangling...).
-
-class Epick
-  : public Filtered_kernel_adaptor<
-               Type_equality_wrapper< Simple_cartesian<double>::Base<Epick>::Type, Epick >,
+constexpr bool epick_use_static_filter =
 #ifdef CGAL_NO_STATIC_FILTERS
-               false >
+    false;
 #else
-               true >
+    true;
 #endif
+
+#ifndef CGAL_EPICK_SINGLE_PRECISION
+// Here Epick is a class, and Double_precision_epick an alias to it.
+class Epick;
+class Single_precision_epick;
+using Double_precision_epick = Epick;
+#else // CGAL_EPICK_SINGLE_PRECISION
+// Here, Epick is an alias to Single_precision_epick.
+class Single_precision_epick;
+class Double_precision_epick;
+using Epick = Single_precision_epick;
+#endif // CGAL_EPICK_SINGLE_PRECISION
+
+namespace internal {
+
+  // Basic objects, constructions, and predicates, using `double` for `FT`.
+  using Double_precision_epick_base =
+      Simple_cartesian<double>::Base<Double_precision_epick>::Type;
+
+  // Add the type equality property, by changing the objects types
+  using Double_precision_epick_base_with_type_equality =
+      Type_equality_wrapper<Double_precision_epick_base, Double_precision_epick>;
+
+  // Change the predicates to filtered predicates (with static filters or not)
+  using Double_precision_epick_with_filtered_predicates =
+      Filtered_kernel_adaptor<Double_precision_epick_base_with_type_equality, epick_use_static_filter>;
+
+  // Same, but with objects using `float` for FT
+  using Single_precision_epick_base =
+      Simple_cartesian<float>::Base<Single_precision_epick>::Type;
+  using Single_precision_epick_base_with_type_equality =
+      Type_equality_wrapper<Single_precision_epick_base, Single_precision_epick>;
+  using Single_precision_epick_with_filtered_predicates =
+      Filtered_kernel_adaptor<Single_precision_epick_base_with_type_equality, epick_use_static_filter>;
+};
+
+#ifndef CGAL_EPICK_SINGLE_PRECISION
+// The following is equivalent to Filtered_kernel< Simple_cartesian<double> >,
+// but it's shorter in terms of template name length (for error messages, mangling...).
+class Epick : public internal::Double_precision_epick_with_filtered_predicates
 {};
+#else // CGAL_EPICK_SINGLE_PRECISION
+class Double_precision_epick : public internal::Double_precision_epick_with_filtered_predicates
+{};
```

Comment:
Please also review this new way to define `CGAL::Epick`:
  - if `CGAL_EPICK_SINGLE_PRECISION` is **not  defined** then:
    - `CGAL::Epick` is a class,
    - `CGAL::Simple_precision_epick` is a class, and
    - `CGAL::Double_precision_epick` is an alias for CGAL::Epick,
  - but if `CGAL_EPICK_SINGLE_PRECISION` **is defined** then:
    - `CGAL::Simple_precision_epick` is a class,
    - `CGAL::Double_precision_epick` is a class, and
    - `CGAL::Epick` is an alias to `CGAL::Simple_precision_epick`. That allows to see explictely that `Epick` is not really the usual one, in the compiler diagnostic messages, and in symbols names.

I have also added names and comments to the three types that were used in the previous definition of `Epick`:
```c++
class Epick
  : public Filtered_kernel_adaptor<
               Type_equality_wrapper< Simple_cartesian<double>::Base<Epick>::Type, Epick >,
               true >
{};
```

In my opinion, that is clearer that way, but you may disagree. Please comment.